src/transformers/models/parakeet/modular_parakeet.py

```diff
@@ -0,0 +1,593 @@
+
+    Main differences:
+    - Parakeet introduces a subsampling layer.
+    - Using Linear layers instead of Conv1d for the feed forward layers in the individual Encoder blocks
+    - Implementation differences in the multi-head attention component.
+    """
+)
+class ParakeetEncoder(ParakeetPreTrainedModel):
+    def __init__(self, config: ParakeetEncoderConfig):
+        super().__init__(config)
+        self.config = config
+        self.gradient_checkpointing = False
+
+        self.dropout = config.dropout
+        self.dropout_positions = config.dropout_positions
+        self.layerdrop = config.layerdrop
+
+        self.input_scale = math.sqrt(config.hidden_size) if config.scale_input else 1.0
+        self.subsampling = ParakeetEncoderSubsamplingConv2D(config)
+        self.encode_positions = ParakeetEncoderRelPositionalEncoding(config)
+
+        self.layers = nn.ModuleList(
+            [ParakeetEncoderBlock(config, layer_idx) for layer_idx in range(config.num_hidden_layers)]
+        )
+
+        self.post_init()
+
+    @auto_docstring
+    @check_model_inputs
+    @can_return_tuple
+    def forward(
+        self,
+        input_features: torch.Tensor,
+        attention_mask: Optional[torch.Tensor] = None,
+        **kwargs: Unpack[TransformersKwargs],
+    ) -> BaseModelOutput:
+        hidden_states = self.subsampling(input_features, attention_mask)
+        hidden_states = hidden_states * self.input_scale
+        position_embeddings = self.encode_positions(hidden_states)
+
+        hidden_states = nn.functional.dropout(hidden_states, p=self.dropout, training=self.training)
+        position_embeddings = nn.functional.dropout(
+            position_embeddings, p=self.dropout_positions, training=self.training
+        )
+
+        if attention_mask is not None:
+            attention_mask = self._get_output_attention_mask(attention_mask, target_length=hidden_states.shape[1])
+
+            # NOTE: @eustlb, which mask utils to do the same?
+            # @ebezzam could use `create_causal_mask` but more complicated
+            # as `and_mask_function` requires callable mask function
+            attention_mask = attention_mask.unsqueeze(1).expand(-1, hidden_states.shape[1], -1)
+            attention_mask = attention_mask & attention_mask.transpose(1, 2)
+            attention_mask = attention_mask.unsqueeze(1)
+            # attention_mask = create_causal_mask(
+            #     self.config,
+            #     hidden_states,
+            #     attention_mask,
+            #     cache_position=torch.arange(attention_mask.shape[1], device=hidden_states.device),
+            #     past_key_values=None,
+            #     position_ids=None,
+            #     # and_mask_function=None,
+            # )
+            # # ensure boolean mask because of `~attention_mask` in later processing
+            # if attention_mask is not None and attention_mask.dtype != torch.bool:
+            #     attention_mask = attention_mask > 0
```

Comment:
@ArthurZucker `create_causal_mask` seems to be clunkier to use than their custom processing, because of this AND operation on 430. 

Seems like `create_causal_mask` has an `and_mask_function` but it has to be a callable function